src/Microsoft.Framework.Runtime/Servicing/ServicingTable.cs

```diff
@@ -20,39 +22,62 @@ public static class ServicingTable
             string assetPath,
             out string replacementPath)
         {
-            return LoadIndex().TryGetReplacement(packageId, packageVersion, assetPath, out replacementPath);
+            var compositeIndex = LoadIndex();
+
+            foreach (var index in compositeIndex)
+            {
+                if (index.TryGetReplacement(packageId, packageVersion, assetPath, out replacementPath))
+                {
+                    return true;
+                }
+            }
+
+            replacementPath = null;
+            return false;
         }
 
-        private static ServicingIndex LoadIndex()
+        private static List<ServicingIndex> LoadIndex()
         {
             return LazyInitializer.EnsureInitialized(ref _index, ref _indexInitialized, ref _indexSync, () =>
             {
-                var index = new ServicingIndex();
-                var runtimeServicing = Environment.GetEnvironmentVariable(EnvironmentNames.Servicing);
-                if (string.IsNullOrEmpty(runtimeServicing))
-                {
-                    var servicingRoot = Environment.GetEnvironmentVariable("PROGRAMFILES(X86)");
-                    if (string.IsNullOrEmpty(servicingRoot))
-                    {
-                        Environment.GetEnvironmentVariable("PROGRAMFILES");
-                    }
+                var compositeIndex = new List<ServicingIndex>();
 
-                    if (string.IsNullOrEmpty(servicingRoot))
+                foreach (var servicingRoot in GetServicingRoots())
+                {
+                    var index = new ServicingIndex();
+                    if (servicingRoot != null)
                     {
-                        // Nothing to do, we don't have Program Files. Just return the uninitialized index.
-                        return index;
+                        index.Initialize(servicingRoot);
```

Comment:
Not because of this change but I realized that we have a bug here. If the path that you pass through the environment variable is not a valid path, the `ServicingIndex` throws. Here: https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Framework.Runtime/Servicing/ServicingIndex.cs#L19

We should check that's a valid path.